packages/flutter/lib/src/material/text_theme.dart

```diff
@@ -0,0 +1,536 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/painting.dart';
+
+import 'typography.dart';
+
+/// Material design text theme.
+///
+/// Definitions for the various typographical styles found in material design
+/// (e.g., button, caption). Rather than creating a [TextTheme] directly,
+/// you can obtain an instance as [Typography.black] or [Typography.white].
+///
+/// To obtain the current text theme, call [Theme.of] with the current
+/// [BuildContext] and read the [ThemeData.textTheme] property.
+///
+/// The Material Design typography scheme was signficantly changed in the
+/// current (2018) version of the specification
+/// (https://material.io/design/typography).
+///
+/// The 2018 spec has thirteen text styles:
+/// ```
+/// NAME         SIZE  WEIGHT  SPACING
+/// headline1    96.0  light   -1.5
+/// headline2    60.0  light   -0.5
+/// headline3    48.0  normal   0.0
+/// headline4    34.0  normal   0.25
+/// headline5    24.0  normal   0.0
+/// headline6    20.0  medium   0.15
+/// subtitle1    16.0  normal   0.15
+/// subtitle2    14.0  medium   0.1
+/// paragraph1   16.0  normal   0.5
+/// paragraph2   14.0  normal   0.25
+/// button       14.0  medium   0.75
+/// caption      12.0  normal   0.4
+/// overline     10.0  normal   1.5
+///
+/// ```
+/// Where "light" is `FontWeight.w300`, "normal" is `FontWeight.w400` and
+/// "medium" is `FontWeight.w500`. The spec calls 'paragraph1' and
+/// 'paragraph2', 'body1' and 'body2'. They've been changed here to
+/// make it possible to support both the old text style names and the
+/// new ones.
+///
+/// Each of the [TextTheme] text styles corresponds to one of the
+/// styles from the 2018. By default, the font sizes, font weights
+/// and letter spacing have not changed.
```

Comment:
have not changed from what? you never mention the earlier spec, just that the new spec "changed".

If you use the text I suggested above, you'll have to tweak this to read better too.

Reply:
OK, I went with:
```
/// Each of the [TextTheme] text styles corresponds to one of the
/// styles from 2018 spec. By default, the font sizes, font weights
/// and letter spacings have not changed from their original,
/// 2014, values.
```